src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                    {
+                        // For GraphQL, insert operation corresponds to Create action.
+                        throw new DataApiBuilderException(
+                            message: $"Could not insert row with given values in the linking table",
+                            statusCode: HttpStatusCode.InternalServerError,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.DatabaseOperationFailed);
+                    }
+                }
+
+                // Process referencing relationships
+                foreach (Tuple<string, object?> referencingRelationship in multipleCreateStructure.ReferencingRelationships)
+                {
+                    string relatedEntityName = GetRelatedEntityNameInRelationship(entity, referencingRelationship.Item1);
+                    MultipleCreateStructure referencingRelationshipMultipleCreateStructure = new(entityName: relatedEntityName,
+                                                                                                 higherLevelEntityName: entityName,
+                                                                                                 inputMutParams: referencingRelationship.Item2,
+                                                                                                 isLinkingTableInsertionRequired: GraphQLUtils.IsMToNRelationship(entity, referencingRelationship.Item1));
+                    IValueNode node = GraphQLUtils.GetFieldNodeForGivenFieldName(parameterNodes, referencingRelationship.Item1);
+
+                    // Many-Many relationships are marked as Referencing relationships because the linking table insertion can happen only when records have been successfully created in both the
+                    // entities involved in the relationship.
+                    // In M:N relationship, both the entities are referenced entities and the linking table is the referencing table. 
+                    // So, populating referencing fields is performed only for 1:N relationships.    
+                    if (currentEntityRelationshipMetadata is not null
+                        && currentEntityRelationshipMetadata.TargetEntityToFkDefinitionMap.TryGetValue(relatedEntityName, out List<ForeignKeyDefinition>? referencingEntityFKDefinitions)
+                        && !referencingEntityFKDefinitions.IsNullOrEmpty())
+                    {
+                        ForeignKeyDefinition referencingEntityFKDefinition = referencingEntityFKDefinitions[0];
+
+                        if (referencingRelationshipMultipleCreateStructure.IsLinkingTableInsertionRequired)
+                        {
+                            PopulateReferencingFields(sqlMetadataProvider, referencingRelationshipMultipleCreateStructure, referencingEntityFKDefinition, multipleCreateStructure.CurrentEntityCreatedValues, isLinkingTable: true, entityName);
```

Comment:
split long line/method call. Helpful to use parameter names as well for such long list of params.